### PR TITLE
logging: unify debug log message format

### PR DESCRIPTION
**Pull request by miampf**
### Context

The current debug messages all have different formats. For example, some use `%s` while others use `%q` for strings. This PR unifies those formats and removes most uses of `%v` (and similair format strings for structs) in exchange for better logging.

### Proposed change(s)

- Use `%q` instead of `%s` for all strings in debug messages
- When printing flags, don't use `%+v`. Instead, log the important flags individually.
- When debug logging structs focus on the important values of the struct instead of printing it with `%+v` or something similair.

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

File: internal/constellation/kubecmd/kubecmd.go [hunk 1]
```diff
@@ -266,7 +266,7 @@ func (k *KubeCmd) ExtendClusterConfigCertSANs(ctx context.Context, alternativeNa
 		k.log.Debug("No new SANs to add to the cluster's apiserver SAN field")
 		return nil
 	}
-	k.log.Debug(fmt.Sprintf("Extending the cluster's apiserver SAN field with the following SANs: %s\n", strings.Join(missingSANs, ", ")))
+	k.log.Debug(fmt.Sprintf("Extending the cluster's apiserver SAN field with the following SANs: %q\n", strings.Join(missingSANs, ", ")))
```
Comment (daniel-weisse) [inline internal/constellation/kubecmd/kubecmd.go:269]:
```suggestion
	k.log.Debug(fmt.Sprintf("Extending the cluster's apiserver SAN field with the following SANs: %q", strings.Join(missingSANs, ", ")))
```

File: internal/constellation/applyinit.go [hunk 2]
```diff
@@ -85,12 +85,17 @@ func (a *Applier) Init(
...
 	}
 
 	// Perform the RPC
-	a.log.Debug(fmt.Sprintf("Making initialization call, doer is %+v", doer))
+	a.log.Debug(fmt.Sprintf(
+		`Making initialization call, doer has the following parameters:
+  endpoint: %q
+  req.KmsUri: %q
+  req.StorageUri: %q
+`, doer.endpoint, doer.req.KmsUri, doer.req.StorageUri))
```
Comment (daniel-weisse) [inline internal/constellation/applyinit.go:98]:
```suggestion
	a.log.Debug("Making initialization call", "endpoint", doer.endpoint, "kmsUri", doer.req.KmsUri, "storageUri")
```

We should just use log attributes to log this information

File: internal/api/versionsapi/cli/rm.go [hunk 3]
```diff
@@ -631,7 +648,7 @@ func (a *azureClient) deleteImage(ctx context.Context, image string, dryrun bool
 	}
 
 	if dryrun {
-		log.Debug(fmt.Sprintf("DryRun: delete image %v", azImage))
+		log.Debug(fmt.Sprintf("DryRun: delete image with gallery %q, image definition %q, resource group %q and version %q", azImage.gallery, azImage.imageDefinition, azImage.resourceGroup, azImage.version))
```
Comment (daniel-weisse) [inline internal/api/versionsapi/cli/rm.go:651]:
```suggestion
		log.Debug(fmt.Sprintf("DryRun: delete image: resource group: %q, gallery: %q, image definition: %q, version: %q"", azImage.resourceGroup, azImage.gallery, azImage.imageDefinition, azImage.version))
```

nit: adhere to the log format of the other message later on in this function

File: e2e/malicious-join/malicious-join.go [hunk 4]
```diff
@@ -172,7 +172,11 @@ func (j *maliciousJoiner) join(ctx context.Context) (*joinproto.IssueJoinTicketR
 		IsControlPlane:     false,
 	}
 	res, err := protoClient.IssueJoinTicket(ctx, req)
-	j.logger.Debug(fmt.Sprintf("Got join ticket response: %+v", res))
+	j.logger.Debug(fmt.Sprintf(
+		`Got join ticket response:
+  ApiServerEndpoint: %q
+  KubernetesVersion: %q
+`, res.ApiServerEndpoint, res.KubernetesVersion))
```
Comment (daniel-weisse) [inline e2e/malicious-join/malicious-join.go:179]:
```suggestion
	j.logger.Debug("Got join ticket response", "apiServerEndpoint", res.ApiServerEndpoint, "kubernetesVersion", res.KubernetesVersion)
```

should just use attributes

**Review by daniel-weisse — COMMENTED** [review]
Overall looks good to me.
Two issues I have with the changes:

1. Custom formatting of log messages.

This mostly happens when logging flags.
I don't think we should introduce custom formatting as this may break log processing and also breaks the "one message per line" we currently have for debug messages.
IMO, we should just use log attributes for all of the cases in this PR.
I annotated the code parts where this happened outside of flag printing.
Flag printing should also use attributes similar to my annotations, e.g.
```Go
cfm.log.Debug("Using flags", "insecure", cfm.flags.insecure, "measurementsURL", cfm.flags.measurementsURL, "signatureURL", cfm.flags.signatureURL)
```

2. Measurement printing

I personally think the few cases we do print measurements as part of debug logging are helpful and should stay in the code.
Since we do this more than once, adding a simple `.String()` method to the `measurements.M` type would be helpful.
I'm also ok with keeping the `%v` formatting for these parts, as the output was readable enough for their use case,

File: internal/attestation/measurements/measurements.go [hunk 5]
```diff
@@ -330,6 +330,15 @@ func (m *M) UnmarshalYAML(unmarshal func(any) error) error {
 	return nil
 }
 
+// String returns a string representation of the measurements.
+func (m M) String() string {
+	returnString := ""
+	for i, measurement := range m {
+		returnString += fmt.Sprintf("%d: 0x%s\n", i, hex.EncodeToString(measurement.Expected))
+	}
+	return returnString
```
Comment (daniel-weisse) [inline internal/attestation/measurements/measurements.go:339]:
```suggestion
	var returnString string
	for i, measurement := range m {
		returnString = strings.Join([]string{returnString, fmt.Sprintf("%d: 0x%s", i, hex.EncodeToString(measurement.Expected)}, ",")
	}
	return returnString
```
As mentioned before, I would prefer to keep the logging to one message per line.

File: joinservice/internal/watcher/validator.go [hunk 6]
```diff
@@ -79,7 +79,7 @@ func (u *Updatable) Update() error {
 	if err != nil {
 		return fmt.Errorf("unmarshaling config: %w", err)
 	}
-	u.log.Debug(fmt.Sprintf("New expected measurements: %+v", cfg.GetMeasurements()))
+	u.log.Debug(fmt.Sprintf("New expected measurements:\n%s", cfg.GetMeasurements().String()))
```
Comment (daniel-weisse) [inline joinservice/internal/watcher/validator.go:82]:
```suggestion
	u.log.Debug(fmt.Sprintf("New expected measurements: %s", cfg.GetMeasurements().String()))
```

File: cli/internal/cmd/configfetchmeasurements.go [hunk 7]
```diff
@@ -152,14 +152,14 @@ func (cfm *configFetchMeasurementsCmd) configFetchMeasurements(
 			return fmt.Errorf("fetching and verifying measurements: %w", err)
 		}
 	}
-	cfm.log.Debug(fmt.Sprintf("Measurements: %#v\n", fetchedMeasurements))
+	cfm.log.Debug(fmt.Sprintf("Measurements:\n%s", fetchedMeasurements.String()))
```
[the remark this answers is not in the document]
Comment (miampf) [reply cli/internal/cmd/configfetchmeasurements.go:155]: Do you want to log all the measurements on the same line? If yes, I would adjust the `.String()` method for `measurements.M` to not use newlines as well.

[the remark this answers is not in the document]
Comment (miampf) [reply cli/internal/cmd/configfetchmeasurements.go:155]: Never mind, I overlooked your other comment.

**Comment by malt3** [comment]
The change looks good to me from a high level.

**Pull request merged**